code/modules/mob/living/silicon/ai/laws.dm

```diff
@@ -17,10 +17,16 @@
 
 	src.laws_sanity_check()
 	src.laws.show_laws(who)
-	if(!everyone)
-		for(var/mob/living/silicon/robot/R in connected_robots)
-			if(R.lawupdate)
-				R.lawsync()
-				R.show_laws()
-				R.law_change_counter++
 
+/mob/living/silicon/ai/post_lawchange(announce = TRUE)
+	. = ..()
+
+	if(!.)
+		return
+
+	addtimer(CALLBACK(src, .proc/lawsync), 0, TIMER_UNIQUE)
+
+/mob/living/silicon/ai/lawsync()
+	for(var/r in connected_robots)
+		var/mob/living/silicon/robot/connected_robot = r
+		connected_robot.lawsync()
```

Comment:
Why does this work?

Reply:
What do you mean?

Reply:
I don't remember, I assume law change counter and show laws are both handled inside lawsync()?

Reply:
Cyborgs' `law_change_counter` isn't used anywhere and currently it increases whenever the Show Laws verb is used. `lawsync` calls `post_lawchange` which calls `show_laws`.